Update automated transfer warnings copy and styles

tellthemachines opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Fixes #100455.

## Proposed Changes

This is a very early iteration on simplifying the warning modals and messages when transferring a site to atomic.
To start with, I've only reduced the copy and updated the background colors. 

<details>
<summary>These are the results</summary>

Activate Hosting Features CTA:

<img width="729" alt="Screenshot 2025-04-07 at 4 43 05 pm" src="https://github.com/user-attachments/assets/4ac41a40-45d0-4a0d-ad33-6d7b32dec48b" />

Plugin details screen CTA AND theme details screen CTA (I updated the title of the plugin details screen to be generic):
 
<img width="687" alt="Screenshot 2025-04-07 at 4 43 35 pm" src="https://github.com/user-attachments/assets/4edff4a3-3e04-46b7-9cfe-4a883ac5a306" />

"Install new theme" from Themes Marketplace:

<img width="768" alt="Screenshot 2025-04-07 at 4 45 14 pm" src="https://github.com/user-attachments/assets/af0125fe-0a33-47b0-a17a-150fd83236a3" />

"Upload" from Plugins Marketplace:

<img width="760" alt="Screenshot 2025-04-07 at 4 45 39 pm" src="https://github.com/user-attachments/assets/182b9e63-cddb-4b5b-8e08-849bc6f63734" />

There's also the possibility that multiple warnings may appear, particularly for legacy sites that may be using unsupported plugins or widgets. In that case, the warnings will look something like this (example modal from plugin details screen):

<img width="684" alt="Screenshot 2025-04-07 at 4 47 27 pm" src="https://github.com/user-attachments/assets/aee090d9-1417-470d-a8d3-dc986d183759" />

</details>

TODO (as per the issue):

* Update the modals to use the Modal from `@wordpress/components`
* Update buttons and badges to use `@wordpress/components`.


## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

1. Test on a site with a business plan that hasn't been transferred to atomic yet (the site URL should be `[sitename].wordpress.com`)
2. Go to the Hosting Overview screen and, under "Hosting Features", click "Activate now" to show the modal;
3. In the Plugins Marketplace, click "Upload" at the top right to navigate to the inline warning;
4. Back in Plugins Marketplace, choose a plugin and click "Install and activate" to show the modal;
5. In the Themes Marketplace, click "Install new theme" at the top right to navigate to the inline warning;
6. Back in the Themes Marketplace, search for and select "Kiosko". Click "Activate" to show the modal.

To test the extra warnings I had to to a code hack. In [this function](https://github.com/Automattic/wp-calypso/blob/trunk/client/state/data-layer/wpcom/sites/automated-transfer/eligibility/index.js#L123), I replaced `data` with the following:
<details>
<summary>alternativeData</summary>

```
const alternativeData = {
			errors: [
				{
					code: 'no_business_plan',
					message: 'Only sites with an eligible plan can be transferred.',
				},
			],
			warnings: {
				plugins: [
					{
						id: 'comment_reply_via_email',
						name: 'Comment Reply via Email',
						description:
							'The ability to reply to comments directly from comment notifications email will no longer be available.',
						supportUrl: 'https://support.wordpress.com/comment-reply-via-email/',
					},
					{
						id: 'writing_helper',
						name: 'Writing Helper',
						description:
							'You will no longer be able to copy posts or seek feedback via the Writing Helper.',
						supportUrl: 'https://support.wordpress.com/writing-helper/',
					},
				],
				widgets: [
					{
						id: 'eventbrite',
						name: 'EventBrite: Event Calendar/Listing',
						description:
							'You will no longer have access to the Eventbrite widget for displaying a calendar of upcoming events.',
						supportUrl:
							'https://en.support.wordpress.com/widgets/eventbrite-event-calendarlisting-widget/',
					},
				],
				subdomain: [
					{
						id: 'wordpress_subdomain',
						name: 'WordPress.com Subdomains',
						description:
							'When you install a plugin, theme, or activate hosting features, <strong>your default site address will change</strong>. We’ll make sure that your site visitors are redirected to the new domain. <a href="https://wordpress.com/support/changing-site-address/#change-a-wpcomstaging-com-address" target="_blank" rel="noreferrer">Learn more ↗</a>',
						domain_names: {
							current: 'somerandomdomainnamefortesting.wordpress.com',
							new: 'somerandomdomainnamefortesting.wpcomstaging.com',
						},
					},
				],
			},
			is_eligible: true,
		};
```

</details>

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

tellthemachines committed:
First pass at updating automated transfer warnings

fditrapani commented:
Thanks for getting this started @tellthemachines. Here are some initial things I found:

**Remove inner padding**
There's an inner padding to the content of the modal that can be removed:
<img width="1840" alt="image" src="https://github.com/user-attachments/assets/18162134-9527-4626-aa16-a767283f955c" />

The text styles can also be improved by not having the text so light. Without even changing the modal, this already looks much better:

<img width="1840" alt="image" src="https://github.com/user-attachments/assets/ec79dd8d-2253-4281-9230-1ef98b85120c" />

Those same updates would work well for the embedded view as well:
<img width="1840" alt="image" src="https://github.com/user-attachments/assets/071809b0-afd5-4e5e-98bb-ecd1fba1a9a8" />

<img width="1840" alt="image" src="https://github.com/user-attachments/assets/b54f4528-a82a-43a3-aab4-f9ac9bd46b2e" />

Unrelated but it's weird how those widths are different for such similar screens 😅

> There's also the possibility that multiple warnings may appear, particularly for legacy sites that may be using unsupported plugins or widgets. In that case, the warnings will look something like this (example modal from plugin details screen)

There's a lot going on in this case. Can we provide something similar to the default view but different due to the additional other changes? 

> For the "when you install a plugin, theme, or activate hosting features..." message that comes from the back end, I think it might be most straightforward to update it to a generic message such as “activating this feature means that…” because the back end doesn't know about the context the warning will appear in.

From the looks of it, it appears to be generic already. I think it's fine to stay with "When you install a plugin, theme, or activate hosting feature, ...." since those are the cases when this would happen. 

> With the modal titles, we should have a bit more latitude as each instance is its own modal, but we could maybe also consider making them all generic (the "Before you continue" seems to work well)

I think Before you continue makes sense and works well in call cases.

tellthemachines committed:
Update text color

tellthemachines committed:
Use Button from WP components

tellthemachines committed:
Update to use WP components modal

tellthemachines commented:
Thanks for the feedback @fditrapani!

I've removed the text color override that was in place so now the default  body `—colour-text` value gets applied.

I tried replacing the dialogs with the WP Modal component. We might need to make some adjustments to it, because its default large size seems a bit too big for its contents:

<img width="1081" alt="Screenshot 2025-04-08 at 5 01 58 pm" src="https://github.com/user-attachments/assets/ead559cd-8d6c-4520-bc33-dd9b6d580017" />


but the `medium` size on the other hand seems a bit too squished:

<img width="1079" alt="Screenshot 2025-04-08 at 5 01 19 pm" src="https://github.com/user-attachments/assets/eee9f522-42b8-4159-8763-69cff5b8690d" />


(I just left whatever padding the component has by default, but that could also be adjusted I guess.)

I also replaced the CTA with a WP Button component, but couldn't replace the badges because the WP Badge component is unfortunately a [private API](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/badge/README.md) 😞 

Let me know your thoughts!

fditrapani commented:
Thanks for the follow up @tellthemachines. It's good to see the right modal in there and for the copy to be more readable. Here are a few things:

Let's go with the medium size modal. It has a better line-length which helps with readability. 

It appears we still have some internal padding on the content. I think it's coming from a card component. When I strip it out, it gives us more room to work with:
<img width="2672" alt="image" src="https://github.com/user-attachments/assets/9d92583d-36e0-4640-9284-607123f61b83" />

Also note that there's something weird going on with the badges. We're setting an artificial height. For the contents of those lines, we would probably be better off using a flex layout vs. absolute positioning. That could help with the wrapping.

Those spacing changes will here too:
<img width="1816" alt="image" src="https://github.com/user-attachments/assets/43d3c67f-abe6-4641-ae22-30fa7a5274fb" />

For this one, it looks like we're treating it a little different than the themes where we only show the banner and then show the URL messaging after you upgrade. I just tested this out and it has me upgrade first, then brings me back here to update the URLs. I think we should follow the same pattern as the themes to simplify the messaging. Here's how that would look with the improved spacing:

<img width="1816" alt="image" src="https://github.com/user-attachments/assets/b6663e6f-2a8b-4baf-a8ab-dc5469eb3bee" />

tellthemachines committed:
Reduce card width and update spacing and alignment

tellthemachines commented:
@fditrapani I reduced the modal width, made adjustments to padding, and reduced the font size slightly. Let me know if that works! 

I also added an ellipsis overflow to the domain names because very long names in the narrow modal would throw off the whole thing.

Before:

<img width="528" alt="Screenshot 2025-04-09 at 2 15 30 pm" src="https://github.com/user-attachments/assets/2d672da5-f97a-41b6-8ee9-d38ea057c921" />

After:

<img width="517" alt="Screenshot 2025-04-09 at 2 15 39 pm" src="https://github.com/user-attachments/assets/95c52d18-d72f-4584-bce6-1e7b32e2e694" />

ramonjd commented:
No action required, just cross posting for discovery.

While testing

- https://github.com/Automattic/wp-calypso/pull/99761

I [noticed](https://github.com/Automattic/wp-calypso/pull/99761#issuecomment-2782161053) that installing themes from Marketplace (/marketplace/theme/) also triggers Atomic transfer, but there's no communication to the user that it's happening.

It'd be handy to have the same messaging if it's at all possible.

tellthemachines commented:
> but there's no communication to the user that it's happening.

the communication would depend on [eligibility warnings](https://github.com/Automattic/wp-calypso/blob/trunk/client/state/data-layer/wpcom/sites/automated-transfer/eligibility/index.js#L94) being present. I'm not sure if they would be on a site with a custom domain? I haven't tested that yet. For sure the warning should pop up if you're just using a default `something.wordpress.com` domain.

ramonjd commented:
> For sure the warning should pop up if you're just using a default something.wordpress.com domain.

Thanks! That was how I was testing it.

fditrapani commented:
Thanks for the updates! this is looking much better and getting closer. Here are a few comments:

**Extra top padding**
It looks like we still have some ghost padding inside the content creating a large gap between the title and the content:

<img width="1285" alt="image" src="https://github.com/user-attachments/assets/c25abb65-46ff-463f-a092-1caa641f2f33" />

<img width="1553" alt="image" src="https://github.com/user-attachments/assets/fd0d1b32-7b76-47d8-bd76-4c2d54c5517d" />

Here's how it looks when I remove the padding from both those elements:

<img width="1387" alt="image" src="https://github.com/user-attachments/assets/cb2fd6f5-d774-4c32-b315-3f305144e6ac" />

**Space between content and buttons**

As you can see the space between the content and buttons is too tight now. Matching the margins around the top + bottom of the URLs will help with that:

<img width="1387" alt="image" src="https://github.com/user-attachments/assets/a776793a-38be-4eb0-a85f-43dd768ced17" />


I adjusted the margin to 24px to fit with our spacial grip of 8px (our spacing is measured in increments of 8. It's 8 to ensure that the spacing is noticeable at different increments. We use the spacial grid to create a balanced rhythm in spacing. There's an exception at times of using a half measure of 4px sometimes too).

<img width="1931" alt="image" src="https://github.com/user-attachments/assets/0cc8f54a-a68e-4325-8c4f-5c32f90a324d" />

**Vertically centre badges**
I see you're already using flexbox to vertically centre the badges but it looks like they're a pixel off:
<img width="569" alt="image" src="https://github.com/user-attachments/assets/803ba6a5-5bbf-479b-988b-ffee6eceea2e" />

It's happening because we explicitly set a height and line-height on the badge. If we unset those, it cleans it up:
<img width="1614" alt="image" src="https://github.com/user-attachments/assets/e2b9d310-d65f-48ca-af58-10352e559737" />

The margins/spacing might need a touch of refinement when this component is embedded. Here it is with some tweaks:
<img width="1387" alt="image" src="https://github.com/user-attachments/assets/a0d79b02-4a3b-4418-8d2b-ff209d0e7338" />

> I also added an ellipsis overflow to the domain names because very long names in the narrow modal would throw off the whole thing.

Good call. I know this might get a little trickier but any chance we can do that on the initial part of the domain so that the .wordpress.com / wpcomstaging.com don't get truncated. It might require moving those elements into their own container and setting the ellipse on the name. It would look something like this but with even spacing:

<img width="1387" alt="image" src="https://github.com/user-attachments/assets/ef8af360-6f7e-4a78-94f2-a130e480c0d4" />

tellthemachines committed:
Update ellipsis behaviour

tellthemachines committed:
Adjust margins and paddings

tellthemachines commented:
@fditrapani I think I've addressed all your feedback with the latest changes!

The modal on hosting features screen has some extra text between the URLs and the buttons:

<img width="524" alt="Screenshot 2025-04-10 at 1 18 45 pm" src="https://github.com/user-attachments/assets/d37c182c-9b73-45d7-a7d7-0e2779be5a37" />

I gave it a margin-bottom of 24px to match the spacing above; let me know if that doesn't work.

tellthemachines commented:
I also re-tested the multiple warning scenario, it's looking a bit better now:

<img width="523" alt="Screenshot 2025-04-10 at 1 59 00 pm" src="https://github.com/user-attachments/assets/4ff331b7-1038-4d17-a3ab-6ffa0fb30f5b" />

Though for some reason on some of the screens an additional message appears about upgrading:

<img width="647" alt="Screenshot 2025-04-10 at 2 17 26 pm" src="https://github.com/user-attachments/assets/e8f400b3-70fd-4b25-b282-2a4d2650ae62" />

Because I'm testing this on a simple site with a Business plan, it looks like there's a bug somewhere in the logic that determines whether the plan exists or not at this point. This doesn't happen when there's only a single warning, so I assume the other warnings are somehow triggering the bug. Might be better to look into that separately though 😬

fditrapani commented:
Thanks for the updates. Today is getting ahead of me. I'll try take a look later today or tomorrow first thing!

adamwoodnz commented on client/blocks/eligibility-warnings/style.scss:
```diff
@@ -1,17 +1,20 @@
 .eligibility-warnings {
 	// 16px from card padding + 32 = 48 gap from top|bottom
 	// 24px from card padding + 24 = 48 gab from left|right
```
```suggestion
	// 24px from card padding + 24 = 48 gap from left|right
```

adamwoodnz reviewed (approved):
Looking good!

Tests well for me. I regression tested my recent change to open Odie from the plugins activation dialog and it's still working as expected. I found that I missed a scenario for theme activation, and the modal stays open when it shouldn't, so I'll build a fix on top of this branch. If it's ready in time maybe we can add it to this branch directly.

![Screenshot 2025-04-11 at 2 32 07 PM](https://github.com/user-attachments/assets/a39a5407-cd6d-4b79-b6a6-bd01654c1a3b)

tellthemachines commented:
Thanks for the review @adamwoodnz !

> If it's ready in time maybe we can add it to this branch directly.

I'll wait until @fditrapani has a chance to check it over before merging. Let me know if there's anything I can help with!

adamwoodnz commented:
> Thanks for the review @adamwoodnz !
> 
> > If it's ready in time maybe we can add it to this branch directly.
> 
> I'll wait until @fditrapani has a chance to check it over before merging. Let me know if there's anything I can help with!

I've opened a PR to fix trunk, which will result in conflict for this. I'll leave a comment on here for how this should be updated.

adamwoodnz commented on client/my-sites/themes/atomic-transfer-dialog.tsx:
```diff
@@ -205,7 +208,7 @@ class AtomicTransferDialog extends Component< AtomicTransferDialogProps > {
 					onProceed={ () => this.handleAccept() }
```
After https://github.com/Automattic/wp-calypso/pull/102552 lands we'll need to add:

```suggestion
					onDismiss={ () => this.handleDismiss() }
					onProceed={ () => this.handleAccept() }
```

adamwoodnz replied:
✅ https://github.com/Automattic/wp-calypso/pull/102552 has shipped

adamwoodnz replied:
Oh I see it's already there 🙂

tellthemachines replied:
Yeah I rebased the PR after you merged yours 😄

tellthemachines committed:
Fix typo

fditrapani reviewed (approved):
Nice work on this @tellthemachines it looks great. Two unrelated pieces that would be great to follow up with:

**Content width**
When you upload a theme, the width of the content is too wide:
<img width="1840" alt="image" src="https://github.com/user-attachments/assets/031b08e5-ba32-42d5-a1f8-fe384b639bde" />

This is what we have for plugins:
<img width="1840" alt="image" src="https://github.com/user-attachments/assets/90fa0a0f-abe9-4a48-9d4b-f7ed8d02b39e" />

We should update the themes page to match.

**Support prompt consistency**
We now have some cases that say "Need help? Contact support" and "Need help?". Can we align these? 

**Upsell + Atomic change**
As you pointed out, we have this case where we're showing an upsell inside:
<img width="1840" alt="image" src="https://github.com/user-attachments/assets/50c48285-ec20-4c22-9ff2-43a7a1160851" />

If you press upgrade and continue, it doesn't actually change your URL after checkout. 

On themes, we only show the upsell plugin and then ask you to convert after:

<img width="1840" alt="image" src="https://github.com/user-attachments/assets/74d9c67c-9cbd-4fe7-81dc-e73375add00d" />

To fix this, we can either fix the flow so that it actually converts it to the right url after purchase and update the layout so that the banner appears above the card:
![image](https://github.com/user-attachments/assets/4d3ca5d2-bb68-4d42-86d5-e5015e9fe7a5)

Alternatively, we can do it like themes that show the upgrade notice on its own. Then after you upgrade, we show the atomic updates messaging.

tellthemachines commented:
> **Support prompt consistency**
> We now have some cases that say "Need help? Contact support" and "Need help?". Can we align these?

@adamwoodnz were you planning to pick these up, given you updated the others, or shall I?

adamwoodnz commented:
> > **Support prompt consistency**
> > We now have some cases that say "Need help? Contact support" and "Need help?". Can we align these?
> 
> @adamwoodnz were you planning to pick these up, given you updated the others, or shall I?

The issue I addressed was specifically for the plugin upgrade and activate flow, and I wan't sure if launching Odie was appropriate for all these scenarios, but I see that it is now. I'll make this the standard behaviour.

tellthemachines commented:
> **Content width**
> When you upload a theme, the width of the content is too wide

@fditrapani I've been looking a little deeper and there are more differences between these screens when users are not on a Business plan, so an upgrade prompt is shown.

Themes page just shows the prompt and nothing else:

<img width="985" alt="Screenshot 2025-04-15 at 3 19 20 pm" src="https://github.com/user-attachments/assets/95cf5b06-e1bd-47d0-9beb-af9c176a2207" />

Plugins page shows a different prompt plus the warning about the domains:

<img width="716" alt="Screenshot 2025-04-15 at 3 19 25 pm" src="https://github.com/user-attachments/assets/b79645ac-8daa-476d-9834-2cba6fac96f5" />

🥲 I'll look into resolving these in another PR. My question is: should we make the theme upload page like the plugin one in this respect too? The prompt + warning looks a little messy.
